pkg/controller/vsphere/reconciler_test.go

```diff
@@ -783,8 +784,381 @@ func TestDelete(t *testing.T) {
+					t.Errorf("Expected: %v, got %v", tc.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("reconciler was not expected to return error: %v", err)
+				}
+			}
+		})
+	}
+}
+func TestUpdate(t *testing.T) {
+	model, session, server := initSimulator(t)
+	defer model.Remove()
+	defer server.Close()
+	credentialsSecretUsername := fmt.Sprintf("%s.username", server.URL.Host)
+	credentialsSecretPassword := fmt.Sprintf("%s.password", server.URL.Host)
+
+	password, _ := server.URL.User.Password()
+	namespace := "test"
+	vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+	instanceUUID := "instanceUUID"
+	vm.Config.InstanceUuid = instanceUUID
+
+	credentialsSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			credentialsSecretUsername: []byte(server.URL.User.Username()),
+			credentialsSecretPassword: []byte(password),
+		},
+	}
+
+	vmObj := object.NewVirtualMachine(session.Client.Client, vm.Reference())
+	task, err := vmObj.PowerOn(context.TODO())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cases := []struct {
+		name          string
+		expectedError error
+		providerSpec  vsphereapi.VSphereMachineProviderSpec
+		labels        map[string]string
+	}{
+		{
+			name: "Successfully update machine",
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				Template: vm.Name,
+				Network: vsphereapi.NetworkSpec{
+					Devices: []vsphereapi.NetworkDeviceSpec{
+						vsphereapi.NetworkDeviceSpec{
+							NetworkName: "test",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Fail on invalid missing machine label",
+			labels: map[string]string{
+				machinev1.MachineClusterIDLabel: "",
+			},
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				Template: vm.Name,
+				Network: vsphereapi.NetworkSpec{
+					Devices: []vsphereapi.NetworkDeviceSpec{
+						vsphereapi.NetworkDeviceSpec{
+							NetworkName: "test",
+						},
+					},
+				},
+			},
+			expectedError: errors.New("test: failed validating machine provider spec: test: missing \"machine.openshift.io/cluster-api-cluster\" label"),
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			labels := map[string]string{
+				machinev1.MachineClusterIDLabel: "CLUSTERID",
+			}
+
+			machineScope := machineScope{
+				Context: context.TODO(),
+				machine: &machinev1.Machine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: "test",
+						Labels:    labels,
+						UID:       apimachinerytypes.UID(instanceUUID),
+					},
+				},
+				providerSpec: &tc.providerSpec,
+				session:      session,
+				providerStatus: &vsphereapi.VSphereMachineProviderStatus{
+					TaskRef: task.Reference().Value,
+				},
+				client: fake.NewFakeClientWithScheme(scheme.Scheme, &credentialsSecret),
+			}
+
+			reconciler := newReconciler(&machineScope)
+
+			if tc.labels != nil {
```

Comment:
Can we move this above creating machineScope so that it matches the pattern in the one above? (Or switch the other to look like this, I don't mind either way, just looking for consistency)